Source/Dafny/Resolver.cs

```diff
@@ -11581,6 +11581,25 @@ private class RBranchExpr : RBranch {
           tp, expectExtract ? "and" : "but not");
       }
 
+      var m = ResolveMember(tok, tp, "IsFailure", out _);
+      if (m != null && m.IsGhost) {
+        origReporter.Error(MessageSource.Resolver, m.tok,
+          "The IsFailure member may not be ghost");
+      }
+      m = ResolveMember(tok, tp, "PropagateFailure", out _);
+      if (m != null && m.IsGhost) {
+        origReporter.Error(MessageSource.Resolver, m.tok,
+          "The PropagateFailure member may not be ghost");
+      }
+
+      if (expectExtract&& m != null) {
+        m = ResolveMember(tok, tp, "Extract", out _);
+        if (m.IsGhost) {
+          origReporter.Error(MessageSource.Resolver, m.tok,
+            "The Extract member may not be ghost");
```

Comment:
I suppose this is okay, but it's stricter than necessary (and stricter than I suggested in #859).

Reply:
Relaxed